riscv-debug-spec.tex

```diff
@@ -619,6 +642,34 @@ \subsection{JTAG Connector}
 connector looks as it does in Table~\ref{tab:header}. The function of each pin
 is described in Table~\ref{tab:pinout}.
 
+%MAW: Why do we only have 'RESET', but no 'TRST(n)' signal?
+% See http://openocd.org/doc/html/Reset-Configuration.html
+% Both are useful.
+% One is explicitly ``test logic reset'', which would imply
+% resetting the JTAG state machine and all associated test logic, in this
+% case the Debug Transport Module, synchronizers, and perhaps even
+% the Debug Module itself. It basically does the same thing as getting
+% the JTAG state machine into TEST_LOGIC_RESET state, which is
+% why it is optional.
+%
+% The other is a convenience which allows the debugger to essentially
+% hit the system reset button, which is basically mutually exclusive from
+% what TRST resets.
+%
+% I think a common workflow would be
+% to assert the halt signal in Debug Module,
+% then tell the debugger to pull the 'RESET' signal.
+% The desired effect woudld be to be able to debug the
+% code that executes on external reset.
+%
+% Another common workflow would be to have been debugging for a while,
+% then you disconnect the debugger.
+% Another user connects, and has no idea what state the debug logic
+% is in. They could assert the TRSTn signal (or clock JTAG TAP into
+% the proper state) and then know that all the Debug Transport Module
+% (and Debug Module logic...?) is in a known state.
```

Comment:
I reset bit like that makes a lot of sense. Although the DM must be reset during power-on to avoid accidentally interfering with regular execution. (At the least, the halt/reset signals it controls must be 0 on power on.)